src/cobra/io/mat.py

```diff
@@ -232,18 +342,74 @@ def create_mat_dict(model: Model) -> OrderedDict:
     return mat
 
 
+def mat_annotations(
+    target_list: List[Object], mat_struct: np.ndarray, _d_replace: str = D_MET
+) -> None:
+    """Process mat structure annotations in place.
+
+    Will process mat structured annotations and add them to a list of new entities
+    (metabolites, reactions, genes) in a format based on identifiers.org.
+
+    Parameters
+    ----------
+    target_list: list[cobra.Object]
+        A list of cobra objects, including metabolites, reactions or genes. The
+        annotations will be added to these lists.
+    mat_struct: np.ndarray
+        A darray that includes the data imported from matlab file.
+    _d_replace: str
+        A string that points to the dictionary of converstions between MATLAB and
+        providers. Default D_MET (for metabolite).
+    """
+    annotation_matlab = list(
+        set(mat_struct.dtype.names).intersection(D_REPLACE[_d_replace].keys())
+    )
+    annotation_lists = [[None]] * len(annotation_matlab)
+    annotation_providers = [D_REPLACE[_d_replace][x] for x in annotation_matlab]
+    for i in range(len(annotation_matlab)):
```

Comment:
use more descriptive names for the running variable, and rather make annotation_lists a dict and iterate over the names like

```
annotations = dict.from_keys(annotation_providers, None)
....
for name in annotation_matlab:
....
```